tests/unit/auth/v4/headerAuthCheck.js

```diff
@@ -200,39 +187,19 @@ describe('v4 headerAuthCheck', () => {
                 '8e43e369f48ea6d3a5f1fe518e14',
         }, 'headers', request, headers);
         const clock = lolex.install(1454962445000);
-        headerAuthCheck(alteredRequest, log, err => {
-            clock.uninstall();
-            assert.ifError(err);
-            done();
-        });
+        const res = headerAuthCheck(alteredRequest, log);
+        clock.uninstall();
+        assert.ifError(res.err);
+        done();
     });
 
-    it('should successfully authenticate', done => {
+    it('should successfully return information', done => {
```

Comment:
Do you still need the done?
It seems you are just testing that the auth version is returned.  Is that all we should be testing here?  If so, let's change the should statement to -- it should return the auth version type